src/eduid_common/authn/eduid_saml2.py

```diff
@@ -198,3 +207,46 @@ def authenticate(app, session_info):
     else:
         return user
     return None
+
+
+def saml_logout(current_app: EduIDBaseApp,
+                user: User,
+                location: str) -> Response:
+    """
+    SAML Logout Request initiator.
+    This function initiates the SAML2 Logout request
+    using the pysaml2 library to create the LogoutRequest.
+    """
+    state = StateCache(session)
+    identity = IdentityCache(session)
+
+    client = Saml2Client(current_app.saml2_config, state_cache=state, identity_cache=identity)
+
+    try:
+        subject_id = decode(session['_saml2_session_name_id'])
+    except KeyError:
+        current_app.logger.warning('The session does not contain the subject id for user {user}')
+        session.invalidate()
+        current_app.logger.info(f'Redirection user to user {location}')
+        return redirect(location)
+
+    session.invalidate()
+    current_app.logger.info(f'Invalidated session for {user}')
+
+    logouts = client.global_logout(subject_id)
+    loresponse = list(logouts.values())[0]
+    # loresponse is a dict for REDIRECT binding, and LogoutResponse for SOAP binding
+    if isinstance(loresponse, LogoutResponse):
+        if loresponse.status_ok():
+            location = verify_relay_state(request.form.get('RelayState', location), location)
+            return redirect(location)
+        else:
+            abort(500)
+
+    headers_tuple = loresponse[1]['headers']
+    location = headers_tuple[0][1]
+    current_app.logger.info(
+        f'Redirecting to {location} to finish account termination and logout for user {user}'
```

Comment:
I would prefer something like, "Redirecting {user] to {location} after successful logout".